src/rust/src/contract.rs

```diff
@@ -209,30 +209,17 @@ fn encode_diagnostic_events(events: &Vec<DiagnosticEvent>) -> Vec<RustBuf> {
 
 fn extract_ledger_effects(
     entry_changes: Vec<LedgerEntryChange>,
-) -> (Vec<RustBuf>, Vec<ReadOnlyBump>) {
+) -> Vec<RustBuf> {
     let mut modified_entries = vec![];
-    let mut read_only_bumps = vec![];
 
     for change in entry_changes {
-        if change.read_only {
-            // Only return expiration bumps for read-only entries (if any).
-            if let Some(expiration_change) = change.expiration_change {
-                if expiration_change.new_expiration_ledger > expiration_change.old_expiration_ledger
-                {
-                    read_only_bumps.push(ReadOnlyBump {
-                        ledger_key: change.encoded_key.into(),
-                        min_expiration: expiration_change.new_expiration_ledger,
-                    });
-                }
-            }
-        } else {
+        assert!(!change.read_only);
```

Comment:
This is not correct - there may be an expiration entry change for RO entry (there won't be a separate `LedgerEntryChange` for expiration entries).

Reply:
I think I'm misunderstanding this function then. I thought the `ExpirationEntry` itself would have `LedgerEntryChanges` populated, meaning that every lifetime bump, even if the associated ContractCode/Data entry is in the RO set, would never be a read_only `LedgerEntryChange`?

Reply:
No, I think we should keep expiration changes correlated with their 'main' entries. Otherwise it will be hard to e.g. compute the rent fees. Basically at the `LedgerEntryChange` level we don't really want to expose how we store the expiration changes.